sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/TokenBucket.java

```diff
@@ -0,0 +1,148 @@
+package com.mesosphere.sdk.scheduler;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This class implements a token bucket to limit the rate at which actions may be taken.  The assumption is that clients
+ * will not take actions unless they receive a token via the {@link #tryAcquire()} method.
+ */
+public class TokenBucket {
+    public static final int DEFAULT_CAPACITY = 256;
+    public static final int DEFAULT_INITIAL_COUNT = DEFAULT_CAPACITY;
+    public static final Duration DEFAULT_MIN_ACQUIRE_INTEVAL = Duration.ofSeconds(5);
+    public static final Duration DEFAULT_MIN_INCREMENT_INTERVAL = Duration.ofSeconds(DEFAULT_CAPACITY);
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
+    private final int capacity;
+    private final Duration incrementInterval;
+    private final Duration acquireInterval;
+    private int count;
+    private long lastRevive = 0;
+
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
+    /**
+     * A TokenBucket acts as a rate limiting helper.  Clients should not perform rate limited work without acquiring a
+     * token from the {@link #tryAcquire()} method.
+     * @param initial The initial number of tokens available
+     * @param capacity The maximum number of tokens the bucket may hold
+     * @param incrementInterval The incrementInterval between adding a new token to the bucket
```

Comment:
Missing `minAcquireInterval`

Alternately feel free to just remove the param docs here, as the params are effectively being documented by the builder, arguably making these duplicates

Also may as well rename minAcquireInterval => acquireInterval to match naming elsewhere